cmd/upgrade/cluster/cmd.go

```diff
@@ -135,6 +135,8 @@ func init() {
 		"For Hosted Control Plane, whether the upgrade should cover only the control plane",
 	)
 
+	flags.MarkDeprecated("control-plane", "Flag is being deprecated, has no function")
```

Comment:
I would do more something like 
"Flag is deprecated and it can be safely omitted. It will be removed in a future version"

But I let you decide on this